crates/tracing/src/lib.rs

```diff
@@ -17,120 +44,181 @@
-    );
-    let layer = tracing_subscriber::fmt::layer()
-        .with_ansi(false)
-        .with_writer(writer)
-        .with_filter(filter)
-        .boxed();
-
-    (layer, guard)
+impl Default for RethTracer {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
-/// A worker guard returned by [`file()`].
+///  Configuration for a logging layer.
 ///
-/// When a guard is dropped, all events currently in-memory are flushed to the log file this guard
-/// belongs to.
-pub type FileWorkerGuard = tracing_appender::non_blocking::WorkerGuard;
+///  This struct holds configuration parameters for a tracing layer, including
+///  the format, filtering directives, optional coloring, and directive.
+#[derive(Debug, Clone)]
+pub struct LayerInfo {
+    format: LogFormat,
+    filters: String,
+    directive: Directive,
+    color: Option<String>,
+}
 
-/// Builds a new tracing layer that writes events to journald.
-///
-/// The events are filtered by `filter`.
+impl LayerInfo {
+    ///  Constructs a new `LayerInfo`.
+    ///
+    ///  # Arguments
+    ///  * `format` - Specifies the format for log messages. Possible values are:
+    ///      - `LogFormat::Json` for JSON formatting.
+    ///      - `LogFormat::LogFmt` for logfmt (key=value) formatting.
+    ///      - `LogFormat::Terminal` for human-readable, terminal-friendly formatting.
+    ///  * `filters` - Additional filtering parameters as a string.
+    ///  * `directive` - Directive for filtering log messages.
+    ///  * `color` - Optional color configuration for the log messages.
+    pub fn new(
+        format: LogFormat,
+        filters: String,
+        directive: Directive,
+        color: Option<String>,
+    ) -> Self {
+        Self { format, directive, filters, color }
+    }
+}
+impl Default for LayerInfo {
+    ///  Provides default values for `LayerInfo`.
+    ///
+    ///  By default, it uses terminal format, INFO level filter,
+    ///  no additional filters, and no color configuration.
+    fn default() -> Self {
+        Self {
+            format: LogFormat::Terminal,
+            directive: LevelFilter::INFO.into(),
+            filters: "debug".to_string(),
+            color: Some("always".to_string()),
+        }
+    }
+}
+
+/// Trait defining a general interface for logging configuration.
 ///
-/// If the layer cannot connect to journald for any reason this function will return an error.
-pub fn journald<S>(filter: EnvFilter) -> std::io::Result<BoxedLayer<S>>
-where
-    S: Subscriber,
-    for<'a> S: LookupSpan<'a>,
-{
-    Ok(tracing_journald::layer()?.with_filter(filter).boxed())
+/// The `Tracer` trait provides a standardized way to initialize logging configurations
+/// in an application. Implementations of this trait can specify different logging setups,
+/// such as standard output logging, file logging, journald logging, or custom logging
+/// configurations tailored for specific environments (like testing).
+pub trait Tracer {
```

Comment:
this is nice. I like that we can have a separate tracer for tests. Let's do it in another PR, so we can merge this one first.